tests/integration/conftest.py

```diff
@@ -0,0 +1,60 @@
+#  This file is part of craft-application.
+#
+#  Copyright 2023 Canonical Ltd.
+#
+#  This program is free software: you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License version 3, as
+#  published by the Free Software Foundation.
+#
+#  This program is distributed in the hope that it will be useful, but WITHOUT
+#  ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+#  SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the GNU Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Configuration for craft-application integration tests."""
+import os
+import pathlib
+import sys
+import tempfile
+
+import pytest
+from craft_application.services import provider
+from craft_providers import lxd, multipass
+
+
+def pytest_configure(config: pytest.Config):
+    config.addinivalue_line("markers", "multipass: tests that require multipass")
+    config.addinivalue_line("markers", "lxd: tests that require lxd")
+
+
+def pytest_runtest_setup(item: pytest.Item):
+    if sys.platform != "linux":
+        for _ in item.iter_markers("lxd"):
+            pytest.skip("lxd tests only run on Linux")
+    elif not lxd.is_installed():
+        for _ in item.iter_markers("lxd"):
+            pytest.skip("lxd not installed")
+
+    if not multipass.is_installed():
+        for _ in item.iter_markers("multipass"):
+            pytest.skip("multipass not installed")
+
+
+@pytest.fixture()
+def provider_service(app_metadata, fake_project):
+    """Provider service with install snap disabled for integration tests"""
+    return provider.ProviderService(app_metadata, fake_project, install_snap=False)
+
+
+@pytest.fixture()
+def provider_tmp_path():
+    """A temporary path accessible to craft providers."""
+    directory = os.getenv("XDG_RUNTIME_DIR")
+    with tempfile.TemporaryDirectory(
+        prefix="craft-application-test-",
+        suffix=".tmp",
+        dir=directory or pathlib.Path.home(),
```

Comment:
can we use `tmp_path`?

Reply:
Sadly no, as multipass doesn't have access to `/tmp`. This is basically an expanded/improved version of [what's in craft-providers](https://github.com/canonical/craft-providers/blob/main/tests/integration/conftest.py#L54).

I've added a fallback for non-Linux platforms as well as expanding the explanation.